blas/gonum/level1cmplx128.go

```diff
@@ -317,3 +317,131 @@ func (Implementation) Zdotu(n int, x []complex128, incX int, y []complex128, inc
 	}
 	return c128.DotuInc(x, y, uintptr(n), uintptr(incX), uintptr(incY), uintptr(ix), uintptr(iy))
 }
+
+// Zdscal scales the vector x by a real scalar alpha.
+// Zdscal has no effect if incX < 0.
+func (Implementation) Zdscal(n int, alpha float64, x []complex128, incX int) {
+	if incX < 1 {
+		if incX == 0 {
+			panic(zeroIncX)
+		}
+		return
+	}
+	if (n-1)*incX >= len(x) {
+		panic(badX)
+	}
+	if n < 1 {
+		if n == 0 {
+			return
+		}
+		panic(negativeN)
+	}
+	if alpha == 0 {
+		if incX == 1 {
+			x = x[:n]
+			for i := range x {
+				x[i] = 0
+			}
+			return
+		}
+		for ix := 0; ix < n*incX; ix += incX {
+			x[ix] = 0
+		}
+		return
+	}
+	if incX == 1 {
+		x = x[:n]
+		for i, v := range x {
```

Comment:
`c128.Scal*` come out much worse here. Having it in assembly would reduce the difference but it would probably still be better to have a specialized `c128.Dscal*`.

On arm64:
```
name                   old time/op  new time/op  delta
ZdscalN10Inc1-4         176ns ± 1%   300ns ± 0%  +71.23%  (p=0.029 n=4+4)
ZdscalN100Inc1-4       1.20µs ± 0%  2.35µs ± 1%  +95.06%  (p=0.029 n=4+4)
ZdscalN1000Inc1-4      11.5µs ± 0%  22.8µs ± 2%  +98.48%  (p=0.029 n=4+4)
ZdscalN10000Inc1-4      117µs ± 1%   230µs ± 1%  +96.42%  (p=0.029 n=4+4)
ZdscalN100000Inc1-4    2.93ms ± 5%  3.33ms ± 1%  +13.80%  (p=0.029 n=4+4)
ZdscalN10Inc10-4        240ns ± 1%   317ns ± 1%  +32.19%  (p=0.029 n=4+4)
ZdscalN100Inc10-4      1.88µs ± 0%  2.45µs ± 0%  +30.79%  (p=0.029 n=4+4)
ZdscalN1000Inc10-4     27.9µs ± 2%  33.3µs ± 0%  +19.37%  (p=0.029 n=4+4)
ZdscalN10000Inc10-4    1.54ms ± 1%  1.47ms ± 2%   -4.82%  (p=0.029 n=4+4)
ZdscalN100000Inc10-4   28.4ms ± 1%  29.0ms ± 1%   +2.10%  (p=0.029 n=4+4)
ZdscalN10Inc1000-4      282ns ± 2%   358ns ± 0%  +26.64%  (p=0.029 n=4+4)
ZdscalN100Inc1000-4    2.26µs ± 0%  2.84µs ± 0%  +25.91%  (p=0.029 n=4+4)
ZdscalN1000Inc1000-4   46.3µs ± 2%  50.4µs ± 1%   +8.96%  (p=0.029 n=4+4)
ZdscalN10000Inc1000-4  3.56ms ± 2%  3.57ms ± 1%     ~     (p=1.000 n=4+4)
```